app/src/main/java/com/android/sample/ui/createRecipe/AddInstructionStepScreen.kt

```diff
@@ -0,0 +1,195 @@
+ * @param navigationActions Actions for navigating between screens.
+ * @param modifier Modifier to be applied to the content.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AddInstructionStepContent(
+    paddingValues: PaddingValues,
+    createRecipeViewModel: CreateRecipeViewModel,
+    navigationActions: NavigationActions,
+    modifier: Modifier = Modifier
+) {
+  var stepDescription by remember { mutableStateOf("") }
+  var stepTime by remember { mutableStateOf("") }
+  var stepCategory by remember { mutableStateOf("") }
+  var selectedIcon by remember { mutableStateOf<IconType?>(null) }
+  var showError by remember { mutableStateOf(false) }
+
+  Column(
+      modifier =
+          modifier.fillMaxSize().padding(paddingValues).padding(horizontal = HORIZONTAL_PADDING.dp),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.Top) {
+        // Step Label
+        Text(
+            text = stringResource(R.string.step_label),
+            style = Typography.titleLarge,
+            color = MaterialTheme.colorScheme.onPrimary,
+            modifier = Modifier.padding(vertical = 8.dp).testTag("StepLabel"))
+
+        // Container for input fields
+        Box(
+            modifier =
+                Modifier.fillMaxWidth()
+                    .background(color = lightCream, shape = RoundedCornerShape(8.dp))
+                    .padding(16.dp)
+                    .testTag("InputContainer")) {
+              Column {
+                // Row for input fields for time, category, and icon
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.spacedBy(8.dp),
+                    verticalAlignment = Alignment.CenterVertically) {
+                      // Time input field
+                      OutlinedTextField(
+                          value = stepTime,
+                          onValueChange = { stepTime = it },
+                          label = {
+                            Text(stringResource(R.string.time_label), style = Typography.bodySmall)
+                          },
+                          keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+                          colors =
+                              TextFieldDefaults.outlinedTextFieldColors(
+                                  containerColor = Color.White),
+                          modifier = Modifier.weight(1f).heightIn(min = 56.dp).testTag("TimeInput"))
+
+                      // Category input field
+                      OutlinedTextField(
+                          value = stepCategory,
+                          onValueChange = { stepCategory = it },
+                          label = {
+                            Text(
+                                stringResource(R.string.category_label),
+                                style = Typography.bodySmall)
```

Comment:
bodySmall is a bit big for the labels. Consider using labelSmall or labelMedium. For problems like this it might be interesting for us to later implement a helper function, that automatically resizes the text depending on it's space ex: AutoSizeText.

Here we see category is going to the next line.
![image](https://github.com/user-attachments/assets/8ee195e8-bee3-4d23-975c-9782425001d5)

Reply:
For the Moment, we will keep it like this. This issue will be resolved in a separate PR. We took your reveiw into account!!